.github/workflows/CI.yml

```diff
@@ -61,7 +58,7 @@ jobs:
     name: Notify Slack Failure
     needs: test
     runs-on: ubuntu-latest
-    if: github.event == 'schedule'
+    if: always() && github.event_name == 'schedule'
```

Comment:
What does the `always` do?

Reply:
so what happens on the slack thing is that it says it depends on the `test` job to pass (not finish which is what I initially thought). If test fails, then this doesn't run. What `always()` does is says "always run this job after test regardless if it passed or failed" we then add that extra stipulation that it will only run when this pipeline runs from a schedule.